waku/node/v2/wakunode2.nim

```diff
@@ -206,12 +211,25 @@ proc mountFilter*(node: WakuNode) =
   node.switch.mount(node.wakuFilter)
   node.subscriptions.subscribe(WakuFilterCodec, node.wakuFilter.subscription())
 
-proc mountStore*(node: WakuNode) =
+proc mountStore*(node: WakuNode, path = ""): Result[void, WakuNodeError] =
+  var db: DbConn
+  try:
```

Comment:
Is this the semantics we want? What happens if I don't specify or want a db?

Alternatively, we could make this a WARN/ERR and keep operating. Not sure what would be best?

cc @kdeme

Reply:
It would probably be better if the database part and the protocol mounting part has a better separation.
And yes, default it could use a memory db perhaps, and you could provide it with a persistent db instead, so you'd need some abstraction.

Reply:
Exactly, it should be a choice which backend is used and this can be abstracted away.

This is also needed for tests and mocking.

Reply:
sqlite supports in-memory databases - it's an easy way to not introduce lots of unnecessary abstraction

Reply:
In memory or not, doesn't matter. As a light node just requesting messages, I don't want this code to run.